File: algo-linear-search/linearSearch.js
function linearSearch(searchTerm, arr) {
  // Iterate over arr
  for (let elem of arr) {
    // If searchTerm matches elem of arr, return index of elem
    if (searchTerm === elem) {
      return arr.indexOf(elem);
    }
  }

  // If searchTerm is not found in arr
  return undefined;
}

function globalLinearSearch(searchTerm, arr) {
  // Arr to contain indices of found elements
  let indices = [];
  // Iterate over arr
  for (let i = 0; i < arr.length; i++) {
    // If searchTerm matches elem of arr, push to indices arr and
    // continue search
    if (searchTerm === arr[i]) {
      indices.push(i);
    }
  }
  // If indices contains elements, return it; otherwise, return
  // undefined
  if (indices.length > 0) {
    return indices;
  } else {
    return undefined;
  }
}

module.exports = { linearSearch, globalLinearSearch };

File: algo-linear-search-py/linear_search.py
def linear_search(value_to_find, array_to_search_through):
    
    index = 0
    
    for elem in array_to_search_through:
        if (elem == value_to_find):
            return index
        index += 1
    
    return None
    
def linear_search_global(value_to_find, array_to_search_through):
    
    index_list = []
    index = 0
    
    for elem in array_to_search_through:
        if (elem == value_to_find):
            index_list.append(index)
        index += 1
    
    if not index_list:
        return None
    else:
        return index_list
